src/test/java/posters/cucumber/support/OpenPageFlows.java

```diff
@@ -6,60 +6,49 @@
 import com.xceptance.neodymium.util.Neodymium;
 
 import io.cucumber.java.en.Given;
-import io.qameta.allure.Step;
 import posters.pageobjects.pages.browsing.HomePage;
-import posters.pageobjects.pages.browsing.ProductdetailPage;
+import posters.pageobjects.pages.browsing.ProductDetailPage;
 import posters.pageobjects.pages.user.LoginPage;
 import posters.pageobjects.pages.user.RegisterPage;
 
 public class OpenPageFlows
 {
     @Given("^homepage is loaded$")
-    @Step("open home page")
-    public static HomePage homepage()
+    public static HomePage homePage()
     {
         // clear cookies to ensure a new session
         clearBrowserCookies();
         // open home page
         open(Neodymium.configuration().url());
-        HomePage homePage = new HomePage();
-        homePage.isExpectedPage();
-        return homePage;
+        return new HomePage().isExpectedPage();
     };
 
     @Given("^login page is loaded$")
-    @Step("open login page")
     public static LoginPage loginPage()
     {
         // open login page and check for expected page
-        LoginPage loginPage = homepage().userMenu.openLogin();
-        loginPage.isExpectedPage();
-        return loginPage;
+        return homePage().userMenu.openLogin();
     };
 
     @Given("^register page is loaded$")
-    @Step("open register page")
     public static RegisterPage registerPage()
     {
         // open login page and check for expected page
-        RegisterPage registerPage = homepage().userMenu.openRegister();
-        registerPage.isExpectedPage();
-        return registerPage;
+        return homePage().userMenu.openRegister();
     };
 
     @Given("^product page \"([^\"]*)\" is open$")
-    @Step("open product page with cleared cookes")
-    public static ProductdetailPage openProductdetailsPageWithClearedCookes(String url)
+    public static ProductDetailPage openProductDetailsPageWithClearedCookes(String url)
     {
         clearBrowserCookies();
         open(Neodymium.configuration().url() + url);
-        return new ProductdetailPage();
+        return new ProductDetailPage().isExpectedPage();
     }
 
-    @Step("open product page without cleared cookes")
-    public static ProductdetailPage openProductdetailsPage(String url)
+    // TODO check if needed - else delete
```

Comment:
@oomelianchuk Please take care of the TODO

Reply:
The method is called in OrderSupport.openProductPageAndAddItoTheCart, so I just removed TODO